service/worker/scanner/history/scavenger.go

```diff
@@ -256,13 +260,17 @@ func (s *Scavenger) startTaskProcessor(
 							getTaskLoggingTags(err, task)...)
 						continue
 					}
-
+					domainName, err := s.domainCache.GetDomainName(task.domainID)
+					if err != nil {
+						return
+					}
```

Comment:
this is probably incorrect, other branches `continue` and/or push an empty message into the response channel.  and it should probably have a log since it's not being returned.